sdks/python/apache_beam/ml/inference/huggingface_inference.py

```diff
@@ -0,0 +1,439 @@
+      model_uri="bert-base-uncased", model_class=AutoModelForMaskedLM))
+
+  Args:
+    model_uri (str): path to the pretrained model on the hugging face
+      models hub.
+    model_class: model class to load the repository from model_uri.
+    device: For torch tensors, specify device on which you wish to
+      run the model. Defaults to CPU.
+    inference_fn: the inference function to use during RunInference.
+      Default is _run_inference_torch_keyed_tensor or
+      _run_inference_tensorflow_keyed_tensor depending on the input type.
+    load_model_args (Dict[str, Any]): keyword arguments to provide load
+      options while loading from Hugging Face Hub. Defaults to None.
+    inference_args ([Dict[str, Any]]): Non-batchable arguments
+      required as inputs to the model's forward() function. Unlike Tensors in
+      `batch`, these parameters will not be dynamically batched.
+      Defaults to None.
+    min_batch_size: the minimum batch size to use when batching inputs.
+    max_batch_size: the maximum batch size to use when batching inputs.
+    large_model: set to true if your model is large enough to run into
+      memory pressure if you load multiple copies. Given a model that
+      consumes N memory and a machine with W cores and M memory, you should
+      set this to True if N*W > M.
+    kwargs: 'env_vars' can be used to set environment variables
+      before loading the model.
+
+  **Supported Versions:** RunInference APIs in Apache Beam
+  supports transformers>=4.18.0.
+  """
+  def run_inference(
+      self,
+      batch: Sequence[Dict[str, Union[tf.Tensor, torch.Tensor]]],
+      model: Union[AutoModel, TFAutoModel],
+      inference_args: Optional[Dict[str, Any]] = None
+  ) -> Iterable[PredictionResult]:
+    """
+    Runs inferences on a batch of Keyed Tensors and returns an Iterable of
+    Tensors Predictions.
+
+    This method stacks the list of Tensors in a vectorized format to optimize
+    the inference call.
+
+    Args:
+      batch: A sequence of Keyed Tensors. These Tensors should be batchable,
+        as this method will call `tf.stack()`/`torch.stack()` and pass in
+        batched Tensors with dimensions (batch_size, n_features, etc.) into the
+        model's predict() function.
+      model: A Tensorflow/PyTorch model.
+      inference_args: Non-batchable arguments required as inputs to the model's
+        forward() function. Unlike Tensors in `batch`, these parameters will
+        not be dynamically batched
+    Returns:
+      An Iterable of type PredictionResult.
+    """
+    inference_args = {} if not inference_args else inference_args
+    if not self._framework:
+      self._framework = "tf" if isinstance(batch[0], tf.Tensor) else "torch"
+
+    # default is always torch keyed tensor. We check if user has provided their
+    # own or we move to infer it with input type.
+    if self._inference_fn != _run_inference_torch_keyed_tensor:
```

Comment:
we can make the default as None and type to be `Optional[Callable[...]]`.

Let's check if it is not None, then run the custom inference_fn else choose depending on the framework.